src/boot/efi/boot.c

```diff
@@ -43,7 +43,26 @@ enum loader_type {
         LOADER_SECURE_BOOT_KEYS,
 };
 
+typedef struct ConfigEntry ConfigEntry;  /* Forward declaration */
+typedef struct Config Config;            /* Forward declaration */
+typedef struct Menu Menu;            /* Forward declaration */
+
+typedef enum {
+    CONFIG_ENTRY,
+    MENU
+} EntryType;
+
+typedef union {
+    ConfigEntry *config_entry;
+    Menu *menu;
+} EntryData;
+
 typedef struct {
+    EntryType entry_type;
+    EntryData entry_data;
+} Entry;
```

Comment:
coding style. use 8space identation. .editorconfig, the emacs + vim files we have in the tree should get that right btw.